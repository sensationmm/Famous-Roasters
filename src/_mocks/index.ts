import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
