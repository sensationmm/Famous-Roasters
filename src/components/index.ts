import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
