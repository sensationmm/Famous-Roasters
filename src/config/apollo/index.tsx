import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
