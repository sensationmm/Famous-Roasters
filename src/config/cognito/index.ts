import useAuth from './auth.hook'

export { useAuth as useAuth }
