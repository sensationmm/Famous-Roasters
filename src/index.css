@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}