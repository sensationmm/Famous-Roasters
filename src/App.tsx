import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
