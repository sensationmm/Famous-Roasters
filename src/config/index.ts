import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
