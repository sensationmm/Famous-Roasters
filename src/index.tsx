import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
