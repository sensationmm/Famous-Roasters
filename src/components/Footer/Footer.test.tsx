import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { MemoryRouter } from 'react-router-dom'
import { i18n } from 'src/config'

import { Footer } from '.'

describe('Footer component', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <MemoryRouter initialEntries={['/']}>
          <Footer />
        </MemoryRouter>
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it.skip('The user can submit email for newsletter subscription', async () => {
    render(
      <I18nextProvider i18n={i18n}>
        <MemoryRouter initialEntries={['/']}>
          <Footer />
        </MemoryRouter>
      </I18nextProvider>,
    )
    const emailInput = await screen.findByTestId('newsletter-email')
    const emailSubmit = await screen.findByTestId('newsletter-submit')
    expect(emailInput).toBeInTheDocument()
    expect(emailSubmit).toBeInTheDocument()
    fireEvent.click(emailInput)
    fireEvent.change(emailInput, { target: { value: 'max.mustermann@email.com' } })
    fireEvent.click(emailSubmit)
  })
})
