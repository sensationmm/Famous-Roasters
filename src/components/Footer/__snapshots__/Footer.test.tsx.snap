// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Renders correctly 1`] = `
<div>
  <footer
    class="bg-brand-black text-base text-white px-5 lg:px-10 py-12 lg:py-20 z-30"
  >
    <div
      class="w-full max-w-7xl mx-auto mb-8 md:mb-12 lg:mb-16 flex flex-col md:flex-row md:items-center justify-start"
    >
      <a
        href="//www.60beans.com"
      >
        <span
          class="sr-only"
        >
          60beans
        </span>
        <img
          alt="60beans"
          class="h-6"
          src="60beans-light.svg"
        />
      </a>
    </div>
    <div
      class="w-full max-w-7xl mx-auto flex flex-col xl:flex-row justify-between"
    >
      <div
        class="grid grid-cols-1 sm:grid-cols-3 gap-1 grow"
      >
        <ul
          class="mb-14"
        >
          <li
            class="mb-4"
          >
            <a
              href="//shop.60beans.com/catalogue"
              target="_self"
            >
              Shop
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//shop.60beans.com/taste-finder"
              target="_self"
            >
              Tastefinder
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.linkedin.com/jobs/search/?currentJobId=3196130257&f_C=86587560&geoId=92000000"
              target="_blank"
            >
              Karriere bei 60beans
            </a>
          </li>
        </ul>
        <ul
          class="mb-14"
        >
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/uber-uns"
              target="_self"
            >
              Über uns
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/roesterei"
              target="_self"
            >
              Unsere Röster
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//blog.60beans.com"
              target="_self"
            >
              Blog
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/warum-60beans"
              target="_self"
            >
              Für Kaffeeröster
            </a>
          </li>
        </ul>
        <ul
          class="mb-14"
        >
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/legal/datenschutz"
              target="_self"
            >
              Datenschutz
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/legal/impressum"
              target="_self"
            >
              Impressum
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/legal/cookie-policy"
              target="_self"
            >
              Cookie Policy
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/legal/agb"
              target="_self"
            >
              Allgemeine Geschäftsbedingungen
            </a>
          </li>
          <li
            class="mb-4"
          >
            <a
              href="//www.60beans.com/legal/widerrufsrecht"
              target="_self"
            >
              Widerrufsrecht
            </a>
          </li>
        </ul>
      </div>
      <div
        class="mb-16 xl:px-20"
      >
        <p
          class="mb-4 text-base leading-6"
        >
          Follow us
        </p>
        <ul
          class="flex text-brand-grey-whisper"
        >
          <li
            class="mr-3 hover:text-brand-grey-bombay"
          >
            <a
              href="//www.facebook.com/60beans.shop"
              target="_blank"
            >
              <svg
                class="undefined"
                fill="inherit"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  clip-rule="evenodd"
                  d="M16.5 6H13.5C12.9477 6 12.5 6.44772 12.5 7V10H16.5C16.6137 9.99748 16.7216 10.0504 16.7892 10.1419C16.8568 10.2334 16.8758 10.352 16.84 10.46L16.1 12.66C16.0318 12.8619 15.8431 12.9984 15.63 13H12.5V20.5C12.5 20.7761 12.2761 21 12 21H9.5C9.22386 21 9 20.7761 9 20.5V13H7.5C7.22386 13 7 12.7761 7 12.5V10.5C7 10.2239 7.22386 10 7.5 10H9V7C9 4.79086 10.7909 3 13 3H16.5C16.7761 3 17 3.22386 17 3.5V5.5C17 5.77614 16.7761 6 16.5 6Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="none"
                />
              </svg>
            </a>
          </li>
          <li
            class="mr-3 hover:text-brand-grey-bombay"
          >
            <a
              href="//www.instagram.com/60beans/"
              target="_blank"
            >
              <svg
                class="undefined"
                fill="inherit"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  clip-rule="evenodd"
                  d="M16 3H8C5.23858 3 3 5.23858 3 8V16C3 18.7614 5.23858 21 8 21H16C18.7614 21 21 18.7614 21 16V8C21 5.23858 18.7614 3 16 3ZM19.25 16C19.2445 17.7926 17.7926 19.2445 16 19.25H8C6.20735 19.2445 4.75549 17.7926 4.75 16V8C4.75549 6.20735 6.20735 4.75549 8 4.75H16C17.7926 4.75549 19.2445 6.20735 19.25 8V16ZM16.75 8.25C17.3023 8.25 17.75 7.80228 17.75 7.25C17.75 6.69772 17.3023 6.25 16.75 6.25C16.1977 6.25 15.75 6.69772 15.75 7.25C15.75 7.80228 16.1977 8.25 16.75 8.25ZM12 7.5C9.51472 7.5 7.5 9.51472 7.5 12C7.5 14.4853 9.51472 16.5 12 16.5C14.4853 16.5 16.5 14.4853 16.5 12C16.5027 10.8057 16.0294 9.65957 15.1849 8.81508C14.3404 7.97059 13.1943 7.49734 12 7.5ZM9.25 12C9.25 13.5188 10.4812 14.75 12 14.75C13.5188 14.75 14.75 13.5188 14.75 12C14.75 10.4812 13.5188 9.25 12 9.25C10.4812 9.25 9.25 10.4812 9.25 12Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="none"
                />
              </svg>
            </a>
          </li>
          <li
            class="mr-3 hover:text-brand-grey-bombay"
          >
            <a
              href="//www.linkedin.com/company/86587560/"
              target="_blank"
            >
              <svg
                class="undefined"
                fill="inherit"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  clip-rule="evenodd"
                  d="M5 3H19C20.1046 3 21 3.89543 21 5V19C21 20.1046 20.1046 21 19 21H5C3.89543 21 3 20.1046 3 19V5C3 3.89543 3.89543 3 5 3ZM8 18C8.27614 18 8.5 17.7761 8.5 17.5V10.5C8.5 10.2239 8.27614 10 8 10H6.5C6.22386 10 6 10.2239 6 10.5V17.5C6 17.7761 6.22386 18 6.5 18H8ZM7.25 9C6.42157 9 5.75 8.32843 5.75 7.5C5.75 6.67157 6.42157 6 7.25 6C8.07843 6 8.75 6.67157 8.75 7.5C8.75 8.32843 8.07843 9 7.25 9ZM17.5 18C17.7761 18 18 17.7761 18 17.5V12.9C18.0325 11.3108 16.8576 9.95452 15.28 9.76C14.177 9.65925 13.1083 10.1744 12.5 11.1V10.5C12.5 10.2239 12.2761 10 12 10H10.5C10.2239 10 10 10.2239 10 10.5V17.5C10 17.7761 10.2239 18 10.5 18H12C12.2761 18 12.5 17.7761 12.5 17.5V13.75C12.5 12.9216 13.1716 12.25 14 12.25C14.8284 12.25 15.5 12.9216 15.5 13.75V17.5C15.5 17.7761 15.7239 18 16 18H17.5Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="none"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <div
        class="mb-14"
      >
        <div
          class="trustpilot-widget -ml-5"
          data-businessunit-id="62c454f8cb03c9a2269fadb6"
          data-locale="de-DE"
          data-style-height="150px"
          data-template-id="53aa8807dec7e10d38f59f32"
          data-theme="dark"
        >
          <a
            href="https://de.trustpilot.com/review/60beans.com"
            rel="noopener"
            target="_blank"
          >
            Trustpilot
          </a>
        </div>
      </div>
    </div>
    <div
      class="w-full max-w-7xl mx-auto text-center px-2 mb-1 xl:px-10"
    >
      <span
        class="text-center text-base leading-6"
      >
        © 2022 60beans. Alle Rechte vorbehalten.
      </span>
      <br />
      <span
        class="text-center text-base leading-6"
      >
        Mit ❤️ aus Berlin.
      </span>
    </div>
  </footer>
</div>
`;
